### PR TITLE
Use `GraphSAGE` model in Reddit example

### PR DESCRIPTION
**Pull request by LukeLIN-web**

### CODE REVIEW

**Review by rusty1s — COMMENTED** [review]
Thank you for the PR. Currently, the inference pass is approximately four times slower than the existing solution, so there seems to be something fishy going on. Need to understand the differences first before merging :(

File: examples/reddit.py [hunk 1]
```diff
@@ -99,7 +55,8 @@ def train(epoch):
 @torch.no_grad()
 def test():
     model.eval()
-    y_hat = model.inference(data.x, subgraph_loader).argmax(dim=-1)
```
Comment (LukeLIN-web) [inline examples/reddit.py:102]: @rusty1s  This is because the previous version set all data.x in GPU.
And current `GraphSAGE` class  will set all data.x in CPU, `x_all = loader.data.x.cpu() `
When I use `    y_hat = model.inference(data.x.cpu(), subgraph_loader).argmax(dim=-1)` in the previous version(6.96it/s), it is slower as the newer version (7.31it/s).

Comment (Kh4L) [reply examples/reddit.py:102]: This should fix the slow inference #7548

**Review by wsad1 — APPROVED** [review]
Thanks @LukeLIN-web. 
Have you verified that the accuracy metric is same before and after the change?

File: CHANGELOG.md [hunk 2]
```diff
@@ -71,6 +71,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 ### Changed
 
+- Use the pre-defined `GraphSAGE` model in the Reddit example ([#7494](https://github.com/pyg-team/pytorch_geometric/pull/7494))
```
Comment (wsad1) [inline CHANGELOG.md:74]:
```suggestion
```

**Comment by LukeLIN-web** [comment]
> Thanks @LukeLIN-web. Have you verified that the accuracy metric is same before and after the change?

Yeah, the accuracy metric is same after the change. 
Screenshot 2023-07-08 at 9 53 21 PM

File: examples/reddit.py [hunk 3]
```diff
@@ -21,54 +20,11 @@
...
-        pbar.close()
-        return x_all
-
-
-model = SAGE(dataset.num_features, 256, dataset.num_classes).to(device)
+subgraph_loader = NeighborLoader(data, input_nodes=None, num_neighbors=[-1],
+                                 shuffle=False, **kwargs)
+
+model = GraphSAGE(dataset.num_features, 256, num_layers=2,
+                  out_channels=dataset.num_classes).to(device)
```
Comment (rusty1s) [inline examples/reddit.py:27]:
```suggestion
                  out_channels=dataset.num_classes, ).to(device)
```

File: examples/reddit.py [hunk 4]
```diff
@@ -21,54 +20,11 @@
...
-        pbar = tqdm(total=len(subgraph_loader.dataset) * len(self.convs))
-        pbar.set_description('Evaluating')
-
-        # Compute representations of nodes layer by layer, using *all*
-        # available edges. This leads to faster computation in contrast to
-        # immediately computing the final representations of each batch:
-        for i, conv in enumerate(self.convs):
-            xs = []
-            for batch in subgraph_loader:
-                x = x_all[batch.n_id.to(x_all.device)].to(device)
```
Comment (rusty1s) [inline examples/reddit.py:60]: I am still a bit confused about this. Here, in the previous example, we explicitly convert intermediate embeddings to CPU, while we are now storing them on GPU.

File: examples/reddit.py [hunk 5]
```diff
@@ -21,54 +20,15 @@
...
-        return x_all
-
-
-model = SAGE(dataset.num_features, 256, dataset.num_classes).to(device)
+subgraph_loader = NeighborLoader(data, input_nodes=None, num_neighbors=[-1],
+                                 shuffle=False, **kwargs)
+
+model = GraphSAGE(
+    dataset.num_features,
+    256,
```
Comment (rusty1s) [inline examples/reddit.py:28]:
```suggestion
    in_channels=dataset.num_features,
    hidden_channels=256,
```

File: examples/reddit.py (hunk 4 above)
Comment (LukeLIN-web) [reply examples/reddit.py:60]: Do you mean it may not be as fast as the previous example if we store intermediate embeddings to the CPU? Or do we need to maintain the semantics?

Comment (rusty1s) [reply examples/reddit.py:60]: Sorry for late reply. I mean that the previous example stored intermediate embeddings on CPU, and we can only maintain the same runtime if we now store them on GPU. I think we still need to look into what causes the real slowdown.
